### ISSUE TITLE
comma-dangle with spread bug

### ISSUE THREAD

**Issue by eddiemoore** [eslint/eslint#7857]
**Tell us about your environment**

* **ESLint Version:** 3.12.2
* **Node Version:** 6.5.0
* **npm Version:** 3.10.3

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**

```
{
  "extends": "standard",
  "rules": {
    "comma-dangle": ['error', {
      arrays: 'always-multiline',
      objects: 'always-multiline',
      imports: 'always-multiline',
      exports: 'always-multiline',
      functions: 'never',
    }]
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
const example = ({
  one,
  two,
  ...rest, // <- this line causes the issue
} = {}) => {
  
}
```

**What did you expect to happen?**
Expected to pass the lint

**What actually happened? Please include the actual, raw output from ESLint.**
```
/XXX/example.js
  4:10  error  Unexpected trailing comma  comma-dangle

✖ 1 problem (1 error, 0 warnings)
```

**Comment by ljharb** [comment]
👍 array rest args can't have a trailing comma, but object rest args certainly can

**Comment by eddiemoore** [comment]
It used to work in eslint `3.4.0`. Seems like there was a breaking change somewhere along the line

**Comment by not-an-aardvark** [comment]
Thanks for the report, but this is working as intended -- trailing commas after rest properties are a syntax error.

Also see: https://github.com/eslint/eslint/issues/7614, https://github.com/eslint/eslint/issues/7502, https://github.com/eslint/eslint/issues/7463, https://github.com/eslint/eslint/issues/7422, https://github.com/eslint/eslint/issues/7297, and https://github.com/eslint/eslint/pull/7298.

**Comment by eddiemoore** [comment]
Ahh thanks for the clarification

**Comment by ljharb** [comment]
@not-an-aardvark that isn't the case for object rest/spread - this absolutely should be reopened as a bug.

**Comment by ljharb** [comment]
oops, nvm, i misunderstood.
